source/common/event/dispatcher_impl.cc

```diff
@@ -169,15 +169,15 @@ Network::DnsResolverSharedPtr DispatcherImpl::createDnsResolver(
         "Apple's API only allows overriding DNS resolvers via system settings. Delete resolvers "
         "config or disable the envoy.restart_features.use_apple_api_for_dns_lookups runtime "
         "feature.");
-    RELEASE_ASSERT(!use_tcp_for_dns_lookups,
+    RELEASE_ASSERT(!dns_resolver_options.use_tcp_for_dns_lookups(),
                    "using TCP for DNS lookups is not possible when using Apple APIs for DNS "
                    "resolution. Apple' API only uses UDP for DNS resolution. Use UDP or disable "
                    "the envoy.restart_features.use_apple_api_for_dns_lookups runtime feature.");
     return std::make_shared<Network::AppleDnsResolverImpl>(*this, api_.randomGenerator(),
```

Comment:
Note that in this case the new `no_default_search_domain` flag won't be used to initialize the AppleDnsResolver.